src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate from = LocalDate.parse(dateFrom, formatter).minusDays(1);
+        LocalDate till = LocalDate.parse(dateTo, formatter).plusDays(1);
+        String[][] splittedData = new String[data.length][];
+        String[] result = new String[names.length];
+        for (int j = 0; j < names.length; j++) {
+            StringBuilder stringBuilder = new StringBuilder();
+            int totalInRange = 0;
+            stringBuilder.append(names[j]).append(" - ");
+            for (int i = 0; i < data.length; i++) {
+                splittedData[i] = data[i].split(" ");
```

Comment:
```suggestion
                String splittedData = data[i].split(" ");
```
тобі не потрібен двохвимірний масив `String[][] splittedData = new String[data.length][];`